server/storage/consistent_store.go

```diff
@@ -0,0 +1,69 @@
+package storage
+
+import (
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/docker/go/canonical/json"
+	"github.com/docker/notary"
+	"github.com/docker/notary/tuf/data"
+)
+
+// ConsistentMetaStorage wraps a MetaStore in order to walk the TUF tree for GetCurrent in a consistent manner
```

Comment:
Consistent here means regardless of race conditions on read vs write or failed updates leaving artifacts in the database, you will always get the correct piece of metadata. This consistency is only relevant in the context of `GetCurrent` where selecting the `root` (or any other role) based purely on ordering the versions in the storage may return a root that can't actually be reached via a timestamp+snapshot due to race conditions and failure cases.